source/Ship.cpp

```diff
@@ -4128,6 +4128,27 @@ const vector<weak_ptr<Ship>> &Ship::GetEscorts() const
 
 
 
+std::shared_ptr<std::string> Ship::GetLimitedFleetId()
+{
+	return limitedFleetId;
+}
```

Comment:
Havimg a const and nom const return of the same thing is weird
The getter shouldnt modify it, give a ref?

Reply:
No, that's how const and non-const accessors work with shared_ptrs. You don't return a reference or pointer to a shared_ptr; you return another shared_ptr. The shared_ptr equivalent of `const *whatever` is `shared_ptr<const whatever>`.